### PR TITLE
ECOM-5961 Fix course ordering bug

### PR DESCRIPTION
**Pull request by MatthewPiatetsky**
@edx/ecommerce @rlucioni

### CODE REVIEW

**Comment by rlucioni** [comment]
@MatthewPiatetsky relatively minor, but please stick to the commit message convention we've been discussing:

```
Summary

Description

Ticket number
```

We shouldn't be relying on GitHub for storing this information. The context about the generator is also good to include in the commit message.

File: course_discovery/apps/api/tests/test_serializers.py [hunk 1]
```diff
@@ -519,6 +519,14 @@ def get_expected_data(self, program, request):
             'price_ranges': program.price_ranges,
             'subjects': SubjectSerializer(program.subjects, many=True).data,
             'transcript_languages': [serialize_language_to_code(l) for l in program.transcript_languages],
+            'courses': MinimalProgramCourseSerializer(
```
Comment (rlucioni) [inline course_discovery/apps/api/tests/test_serializers.py]: This doesn't need to be here, it's already included in the expected data by the superclass.

File: course_discovery/apps/api/tests/test_serializers.py [hunk 2]
```diff
@@ -670,9 +678,11 @@ def test_course_ordering_with_no_start(self):
 
     def test_data_without_course_sorting(self):
```
Comment (rlucioni) [inline course_discovery/apps/api/tests/test_serializers.py:671]: This test case is really "test_data_with_custom_course_sorting."

**Review by rlucioni — CHANGES_REQUESTED** [review]
@MatthewPiatetsky I see you converting the generator to list repeatedly, but I don't see any new test cases that would have failed without your change. Am I missing something?

**Comment by MatthewPiatetsky** [comment]
@rlucioni I had a bunch of test failures once I changed the serializer, the tests and the serializer were both using program.course_runs, so they both had the incorrect data, also didn't want others to accidentally get the wrong data because they were unintentionally using a generator

File: course_discovery/apps/api/tests/test_serializers.py (hunk 2 above)
Comment (MatthewPiatetsky) [reply course_discovery/apps/api/tests/test_serializers.py:671]: what is the difference?  
From the serializer point of view the regular data is just stored in sorted order

Comment (rlucioni) [reply course_discovery/apps/api/tests/test_serializers.py:671]: That's true. It's not a big deal, I just thought it was a little misleading; the data _is_ sorted, just not by the serializer.

**Comment by rlucioni** [comment]
@MatthewPiatetsky that makes sense, thanks. So the serializer started working right, but the tests were still doing the wrong thing with the generator and failed.

**Review by rlucioni — COMMENTED** [review]
Good with me once you remove the duplicated test data.

**Review by rlucioni — APPROVED** [review]
I meant to approve :|

**Pull request merged**